app/services/operations/mark_as_ready_application.rb

```diff
@@ -0,0 +1,23 @@
+module Operations
+  class MarkAsReadyApplication
+    def initialize(application_id:)
+      @application = CrimeApplication.find(application_id)
+    end
+
+    def call
+      raise Errors::AlreadyMarkedAsReady if application.ready_for_assessment?
+      raise Errors::AlreadyCompleted if application.assessment_completed?
+      raise Errors::AlreadyReturned if application.returned?
```

Comment:
I considered simplifying these errors into one e.g. CanOnlyBeRecieved but I wondered if the ability to differentiate between which status raised the error was going to be helpful down the line.. but open to opinions

Reply:
I think this is good. Probably useful to know what state something was in when the user tried to mark as ready for assessment - would make it easier to debug if ever it did happen